roomserver/internal/perform/perform_peek.go

```diff
@@ -151,11 +151,28 @@ func (r *Peeker) performPeekRoomByID(
 		}
 	}
 
-	// If the server name in the room ID isn't ours then it's a
-	// possible candidate for finding the room via federation. Add
-	// it to the list of servers to try.
+	// handle federated peeks
+	// FIXME: don't create an outbound peek if we already have one going.
 	if domain != r.Cfg.Matrix.ServerName {
```

Comment:
This won't work in cases where you create the room, other servers join then you leave the room then try to peek into it, as this will resolve to `false`. We shouldn't ever be switching on the domain of the room ID like this.